splink/comparison_level_library.py

```diff
@@ -6,7 +6,12 @@
 
 
 class NullLevelBase(ComparisonLevel):
-    def __init__(self, col_name, valid_string_regex: str = None) -> ComparisonLevel:
+    def __init__(
+        self,
+        col_name,
+        valid_string_regex: str = None,
```

Comment:
maybe i've misunderstood, but when you have `invalid_dates_as_null=True` is this not a date-format string, like `"yyyy-MM-dd"` / `"%Y-%m-%d"` etc. rather than regex?

Reply:
Ah yea I understand, that stuff isn't technically regex. Yea I will have a think about the naming to make it clearer